run_xnat2bids.py

```diff
@@ -181,6 +184,76 @@ def fetch_job_ids(stdout):
 
     return jobs
 
+def diff_data_directory(bids_root, user, password):
+
+    missing_sessions = []
+
+    # Establish connection 
+    connection = requests.Session()
+    connection.verify = True
+    connection.auth = (user, password)
+
+    host = "https://xnat.bnc.brown.edu"
+
+    # Gather list of existing projects in data directory
+    projects = [proj.name for proj in os.scandir(bids_root) if proj.is_dir()]
+
+    for project in projects:
+        # Gather list of studies for every project
+        studies = [stu.name.split("-")[1] for stu in os.scandir(f"{bids_root}/{project}")]
+
+        for study in studies:
+            # Request all sessions in PROJECT_STUDY from XNAT.
+            proj_study = f"{project}_{study}".upper()
+            sessions = get_sessions_from_project(connection, host, proj_study)
+            
+            for experiment in sessions:
+                # Get date of most recent change for every session
+                latest_date = experiment['date']
```

Comment:
Are you able to access the "date added" field instead? That should correspond to the date it was synced from the relay or uploaded manually, which may be considerably more recent than the scan date.

Reply:
Originally, this is what I was doing, but I couldn't perceive a scenario where a session with the same name is resynced on top of an existing session on XNAT. I may be wrong in assuming this.  I can make the change to check both "date" and "date added."  I assumed "date added" would never change after the first time its uploaded.
 
If a user would manually uploads session data or resources to a session that already exists on XNAT, the insert_date (date added) field **is not** updated.  Once a user add resources or session data, they can manually update the "date" field by editing the session details. Basically, this checks to see if the user manually added data and updated the "date" field.

Reply:
It happens more often than would be ideal that someone realizes that not all of the data made it for one reason or another (sync failed midway through, they sent the data from the scanner before the data was actually ready to be sent so only some of it went through, etc.). The alternative is just that we tell people that if they want to re-convert data, they have to delete it from their xnat export directory on Oscar first, which would also be ok.